src/dune_file.ml

```diff
@@ -159,17 +162,35 @@ module Pps_and_flags = struct
     let decode =
       let+ l, flags =
         until_keyword "--"
-          ~before:(plain_string (fun ~loc s -> (loc, s)))
-          ~after:(repeat string)
+          ~before:String_with_vars.decode
+          ~after:(repeat String_with_vars.decode)
+      and+ syntax_version = Syntax.get_exn Stanza.syntax
       in
+      let check_allowed_variable flag =
+        if String_with_vars.has_vars flag then
+          Syntax.Error.since (String_with_vars.loc flag)
+            Stanza.syntax
+            (1, 10)
+              ~what:"Using variables in pps flags"
+      in
+      let flags = Option.value flags ~default:[] in
+      if syntax_version < (1, 10) then
+        List.iter ~f:check_allowed_variable flags;
```

Comment:
Small nitpick: I would do the check right at the end, on the final `more_flags @ flags` list. That would make the code slightly shorter.

Reply:
Ok !